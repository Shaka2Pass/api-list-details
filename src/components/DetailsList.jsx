import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const DetailsList = ({ name, image, status, occupation, portrayedby }) => (
  <section>
    <p><Link to={`detail/${name}`}>{name}</Link></p>
    {image && <img src={image} alt="image of xfiles character" />}
    {status && <p>Status:{status}</p>}
    {occupation && <p>Occupation:{occupation}</p>}
    {portrayedby && <p>Actor:{portrayedby}</p>}
  </section>
);

DetailsList.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  portrayedby: PropTypes.string.isRequired
};

export default DetailsList;
