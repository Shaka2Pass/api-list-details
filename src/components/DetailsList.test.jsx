import React from 'react';
import { shallow } from 'enzyme';
import DetailsList from './DetailsList';

describe('DetailsList component', () => {
  it('matches a snapshot of DetailsList', () => {
    const wrapper = shallow(<DetailsList
      name="Dana Scully"
      image="https://vignette.wikia.nocookie.net/x-files/images/3/31/Absalom.jpg/revision/latest/scale-to-width-down/310?cb=20070708201919"
      status="alive"
      occupation="FBI Agent"
      portrayedby="Gillian Anderson"/>);
    expect(wrapper).toMatchSnapshot();
  });
});

