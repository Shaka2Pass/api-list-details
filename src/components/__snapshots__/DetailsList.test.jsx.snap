// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsList component matches a snapshot of DetailsList 1`] = `""`;
